ca_description/urdf/april_tag.xacro

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" ?>
+<robot xmlns:xacro="http://ros.org/wiki/xacro">
+
+  <xacro:macro name="april_tag" params="name_prefix base_link *origin">
+
+    <link name="${name_prefix}_april_tag_link">
+      <visual>
```

Comment:
I just realized about this. Is the visual tag mandatory? Since you only want to apply a texture, do you need to create this visual block?

Reply:
If I delete this, the Apriltag doesn't appear. This xacro corresponds to the Apriltag of the robot. 
How can a texture be applied if there is no visual object to apply it to?